test/test_kmeans.py

```diff
@@ -15,26 +15,19 @@ def test_fails_for_repeating_centroids():
     data = pd.read_csv('/Users/amruk/PycharmProjects/kmeans/setA.txt', delimiter=' ')

```

Comment:
Do not use real data in tests. Especially that you are using absolute path - this will not work anywhere but on your PC.
You have [`make_blobs`](https://scikit-learn.org/stable/modules/generated/sklearn.datasets.make_blobs.html) and [other functions](https://scikit-learn.org/stable/modules/classes.html#samples-generator) for that.